packages/adapter/lib/adapter/adapter.js

```diff
@@ -2589,6 +2589,43 @@ function Adapter(options) {
                 return tools.maybeCallbackWithError(callback, tools.ERRORS.ERROR_DB_CLOSED);
             }
 
+            params = params || {};
+
+            if (design === 'system' && !params.startkey && !params.endkey) {
```

Comment:
A comment referencing the performance boost might be nice :) Just so we know next year why this is being done.

Reply:
Done